Tests/SonarScanner.MSBuild.PreProcessor.Tests/SonarWebServiceTest.cs

```diff
@@ -208,6 +208,13 @@ public void TryGetQualityProfile64()
             result.Item1.Should().BeTrue();
             result.Item2.Should().Be("profile3k");
 
+            //multiple QPs for a project, taking the default one
```

Comment:
@mickael-caro-sonarsource I didn't think it was possible to have multiple QPs for a single project - if it is, then we'll need to change how we handle them in SonarLint. In what circumstances can this happen?

Reply:
This is for very old version of SQ : In 6.3.1 for example, the query we make (api/qp/search?project=xxx) returns as many QPs as you have, even if there are multiple one for the same language.

Reply:
Ah, ok. That's presumably because the parameter name was changed, and v6.3 was expecting the now-deprecated `projectKey` instead, and SQ ignored parameters it didn't recognise.

It seems like a bug in S4MSB to arbitrarily pick the first QP in this case, and will impact the analysis for users with old versions of SQ that use the new version of the scanner; they won't see an error during analysis, but the set of rules used won't be what they expect. This could be pretty hard to diagnose.

Wouldn't it be better to fail the analysis in this case?

Reply:
There's at least a warning in that case, saying that we will take the default QP for that language, but i'll check if we can seek for the associated QP to that project.

Reply:
Looking for the correct QP in that case effectively means supporting the old version of SQ for this specific API call. However, there were a number of other web apis that changed the expected parameter name from `projectKey` to `project` at the same time so there may be other web calls that are silently returning the wrong data.

Fixing this specific case makes it unclear whether older versions of SQ are supported or not.
It might be simpler to throw an exception telling the user the scanner doesn't support their version of SQ and that they should use an older version of the scanner. That, or check that the other API calls made by S4MSB work correctly on older versions of SQ, which is a lot more work.

Reply:
Yes this was indeed to wait a bit before implementing the fail-fast mechanism if version is unsupported, thanks for the input !